### ISSUE TITLE
🐛 no entry found for key

### ISSUE THREAD

**Issue by laptou** [biomejs/biome#8482]
### Environment information

```block
bun biome rage                  
CLI:
  Version:                      2.3.9
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.21.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0
```

### What happened?

1. Run `biome check` on a project containing `worker-configuration.d.ts` generated by Wrangler
2. Receive an error:
```
> bun biome check                 
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_js_semantic/src/semantic_model/scope.rs:129:39
Thread Name: biome::workspace_worker_2
Message: no entry found for key

worker-configuration.d.ts internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: no entry found for key
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

Checked 28 files in 28ms. No fixes applied.
```

Here's a [repro repo](https://github.com/laptou/biome-repro-1765858249810).

### Expected result

It should not throw this error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by laptou** [comment]
This bug does not occur on 2.3.8.

**Comment by laptou** [comment]
guys i promise i searched to see if this issue already existed first
